docs/development.rst

```diff
@@ -0,0 +1,198 @@
+#. `An issue is filed`_ describing the shortcoming, defect, or new area of functionality.
+   The issue describes how the software will be different than it is now and why this is desirable.
+#. Development work proceeds in a branch (typically created from a recent master branch revision).
+   Prior to this development or in parallel with it,
+   communication within the team helps identify and clear roadblocks early.
+#. When development has progressed sufficiently,
+   the branch is `proposed for review`_.
+#. One or more other developers `review the changes`_.
+#. The previous two steps may repeat a number of times until the changes pass review.
+#. Upon passing review,
+   the changes are `merged into the master branch`_.
+
+
+.. _An issue is filed:
+
+Filing Issues
+-------------
+
+The project uses `GitHub issues <https://github.com/leastauthority/magic-folder/issues>`_ for issue tracking.
+
+.. _proposed for review:
+
+Proposing Changes
+-----------------
+
+To propose a change,
+create a `GitHub Pull Request <https://github.com/leastauthority/magic-folder/pulls>`_ against master.
+
+Changes should:
+
+  * have a corresponding ticket.
+  * be in a branch named ``<ticket number>.<some short descriptive text>``.
+  * have appropriate developer-facing and user-facing documentation.
+  * have full line and branch coverage provided by the test suite.
+  * be implemented in a way that is not unnecessarily difficult to understand and maintain.
+  * satisfy all of the mechnical checks performed by the continuous integration system.
+
+News Fragments
+~~~~~~~~~~~~~~
+
+One of the mechnical checks performed by continuous integration against proposed changes is the existence of a "news fragment".
+News fragments are assembled at release time by `towncrier <https://pypi.org/project/towncrier/>`_ to contribute to the release announcement.
+News fragments are meant to be user-facing and should have a consistent style.
+News fragments can use a inline formatting directives from reStructuredText.
+
+magic-folder's news fragment style is adapted from `the style guidelines from the Twisted project>`_.
+The fragment types accepted are canonically defined by the towncrier configuration file in the project root.
+
+Here are a few guidelines which should help you write good news fragments:
+
+  * The entry SHOULD contain a high-level description of the change suitable for end users.
+  * When the changes touch Python code,
+    the grammatical subject of the sentence SHOULD be a Python class/method/function/interface/variable/etc,
+    and the verb SHOULD be something that the object does.
+    The verb MAY be prefixed with "now".
+  * When the changes touch user interface,
+    the grammatical subject of the sentence SHOULD identify the part of the user interface affected
+    and the verb SHOULD be something that the user interface does.
+  * For bugfix,
+    it MAY contain a reference to the version in which the bug was introduced.
```

Comment:
Leading spaces before `*` will make this list render as a blockquote:

https://github.com/LeastAuthority/magic-folder/blob/164.dev-docs-etc/docs/development.rst#id2

Reply:
On GitHub, apparently, but not when I do local renders with Sphinx.  But I guess it also works with Sphinx without the leading spaces, so I'll remove them.  I *hope* readthedocs renders like local Sphinx builds and not like GitHub.  If that's true, we should probably treat readthedocs/Sphinx as the authority.  GitHub doesn't really want us to use rst at all and would greatly prefer "github flavored markdown" for maximum vendor lock-in. :)